### ISSUE TITLE
Error: Failed to download Helm from location https://get.helm.sh/helm-v1.2.1-linux-amd64.zip

### ISSUE THREAD

**Issue by jsjoeio** [Azure/setup-helm#99]
### What happened?

```shell
Run azure/setup-helm@v3.3
Downloading v1.2.1
  Error: Error: Failed to download Helm from location https://get.helm.sh/helm-v1.2.1-linux-amd[6](https://github.com/coder/code-server/actions/runs/3316135595/jobs/5478016380#step:4:7)4.zip
```

### Version

- [X] I am using the latest version

### Runner

Ubuntu 20.04.5

### Relevant log output

```shell
##[debug]Evaluating condition for step: 'Install helm'
##[debug]Evaluating: (success() && (steps.changed-files.outputs.any_changed == 'true'))
##[debug]Evaluating And:
##[debug]..Evaluating success:
##[debug]..=> true
##[debug]..Evaluating Equal:
##[debug]....Evaluating Index:
##[debug]......Evaluating Index:
##[debug]........Evaluating Index:
##[debug]..........Evaluating steps:
##[debug]..........=> Object
##[debug]..........Evaluating String:
##[debug]..........=> 'changed-files'
##[debug]........=> Object
##[debug]........Evaluating String:
##[debug]........=> 'outputs'
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'any_changed'
##[debug]....=> 'true'
##[debug]....Evaluating String:
##[debug]....=> 'true'
##[debug]..=> true
##[debug]=> true
##[debug]Expanded: (true && ('true' == 'true'))
##[debug]Result: true
##[debug]Starting: Install helm
##[debug]Loading inputs
##[debug]Evaluating: secrets.GITHUB_TOKEN
##[debug]Evaluating Index:
##[debug]..Evaluating secrets:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'GITHUB_TOKEN'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Loading env
Run azure/setup-helm@v3.3
  
::group::Downloading v1.[2](https://github.com/coder/code-server/actions/runs/3316135595/jobs/5478071781#step:4:2).1
Downloading v1.2.1
  ##[debug]isExplicit: 1.2.1
  ##[debug]explicit? true
  ##[debug]checking cache: /opt/hostedtoolcache/helm/1.2.1/x64
  ##[debug]not found
  ##[debug]Downloading https://get.helm.sh/helm-v1.2.1-linux-amd64.zip
  ##[debug]Downloading /home/runner/work/_temp/7[3](https://github.com/coder/code-server/actions/runs/3316135595/jobs/5478071781#step:4:3)8081a7-2c00-[4](https://github.com/coder/code-server/actions/runs/3316135595/jobs/5478071781#step:4:4)e8b-8118-a9822f6644a[5](https://github.com/coder/code-server/actions/runs/3316135595/jobs/5478071781#step:4:5)
  ##[debug]Failed to download from "https://get.helm.sh/helm-v1.2.1-linux-amd[6](https://github.com/coder/code-server/actions/runs/3316135595/jobs/5478071781#step:4:6)4.zip". Code(404) Message(Not Found)
  Error: Error: Failed to download Helm from location https://get.helm.sh/helm-v1.2.1-linux-amd64.zip
  ##[debug]Node Action run completed with exit code 1
  ##[debug]Finishing: Install helm
```
https://github.com/coder/code-server/actions/runs/3316135595/jobs/5478071781

**Comment by uchinda-sph** [comment]
facing the same issue

**Comment by ethan-k** [comment]
Faced the same issue but replaced `azure/setup-helm`  with the below code for now 

```
      - name: Install Helm
        run: |
          curl https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash
```

**Comment by dotrieutan** [comment]
got same problem, the download link was updated or something ?

**Comment by jmd9019** [comment]
Getting same problem

**Comment by vdhpieter** [comment]
They relied on the default ordering of the GH query and that probably got changed...

Changing the query used here and loosing the reverse (why do that if the API can do it for you?):
https://github.com/Azure/setup-helm/blob/b70d33f56dce95f7970896b826403d7c1bbe8a22/src/run.ts#L58-L73
```gql
query {      
  repository(name: "helm", owner: "helm") {
    releases(first: 10, orderBy: {field: CREATED_AT, direction: DESC}) {
      nodes {
        tagName
      }
    }
  }
}
```

Maybe I find time to PR this, but if somebody wants to do it be my guest!

**Comment by marcindabrowski** [comment]
But even when an order is added some computation of the latest need to be done. The latest could be some patch for previous major/minor.

**Comment by sverweij** [comment]
@marcindabrowski / @vdhpieter we might want to use the `isLatest` attribute from the GitHub API for this. See - [graphql/reference/objects#release](https://docs.github.com/en/graphql/reference/objects#release).

Related: [recent announcement](https://github.blog/changelog/2022-10-21-explicitly-set-the-latest-release/) that the meaning of the `isLatest` attribute would change a bit.

**Comment by OliverMKing** [comment]
#100 will be merged in and released shortly. That fixes this issue. Thanks everyone for the help!

**Comment by ImRodry** [comment]
The linked PR was merged but this still isn’t resolved, are we waiting for anything else?

**Comment by vdhpieter** [comment]
@ImRodry the PR (https://github.com/Azure/setup-helm/pull/100) is not yet merged

**Comment by ImRodry** [comment]
Ah ok was looking at https://github.com/clusterpedia-io/clusterpedia-helm/pull/15
Do you have an ETA for when that will be merged?

**Comment by OliverMKing** [comment]
This has been released in v3.4 (and v3). Please upgrade to one of these versions. Thanks!

**Comment by Endevir** [comment]
@mprimeaux same thing, it seems that https://get.helm.sh/ is currently unavailable, not related with GH actions

upd. Seems like Azure Front Door is down https://azure.status.microsoft/en-us/status , which is front-end for get.helm.sh as seen in nslookup:

```
nslookup get.helm.sh

Non-authoritative answer:
get.helm.sh	canonical name = get-helm-front-door-cdn-c9drhfggd8ewfzck.z03.azurefd.net.
get-helm-front-door-cdn-c9drhfggd8ewfzck.z03.azurefd.net	canonical name = mr-z03.tm-azurefd.net.
mr-z03.tm-azurefd.net	canonical name = shed.dual-low.part-0017.t-0009.t-msedge.net.
shed.dual-low.part-0017.t-0009.t-msedge.net	canonical name = part-0017.t-0009.t-msedge.net.
Name:	part-0017.t-0009.t-msedge.net
Address: 13.107.246.45
Name:	part-0017.t-0009.t-msedge.net
Address: 13.107.213.45
Name:	part-0017.t-0009.t-msedge.net
Address: 2620:1ec:bdf::45
Name:	part-0017.t-0009.t-msedge.net
Address: 2620:1ec:46::45
```
